WidgetKit/OpenTabs/TabProvider.swift

```diff
@@ -17,32 +18,32 @@ struct TabProvider: TimelineProvider {
     }
 
     func getSnapshot(in context: Context, completion: @escaping (OpenTabsEntry) -> Void) {
-        let allOpenTabs = SimpleTab.getSimpleTabs()
-
-        let openTabs = allOpenTabs.values.filter {
+        let openTabs = SimpleTab.getSimpleTabs().values.filter {
             !$0.isPrivate
         }
 
-        var tabFaviconDictionary = [String: Image]()
         let simpleTabs = SimpleTab.getSimpleTabs()
-        for (_, tab) in simpleTabs {
-            guard !tab.imageKey.isEmpty else { continue }
-            let fetchedImage = FaviconFetcher.getFaviconFromDiskCache(imageKey: tab.imageKey)
-            let bundledFavicon = getBundledFavicon(siteUrl: tab.url)
-            let letterFavicon = FaviconFetcher.letter(forUrl: tab.url ?? URL(string: "about:blank")!)
-            let image = bundledFavicon ?? fetchedImage ?? letterFavicon
-            tabFaviconDictionary[tab.imageKey] = Image(uiImage: image)
+        let siteImageFetcher = DefaultSiteImageFetcher.factory()
+
+        Task {
+            let tabFaviconDictionary = await withTaskGroup(of: (String, SiteImageModel).self,
+                                                           returning: [String: Image].self) { group in
+                for (_, tab) in simpleTabs {
+                    group.addTask {
+                        await (tab.imageKey,
+                               siteImageFetcher.getImage(urlStringRequest: tab.url?.absoluteString ?? "",
```

Comment:
I guess this is my lack of knowledge about this package, but how come we don't do `?? URL(string: "about:blank")`?

Reply:
If it ends up needing to fallback to a letter image it will be set to A if the url is about:blank. I think # is the fallback we have if all else fails in the library which is what an empty string will produce. This should never be nil tho. One of the things I want to look at when refactoring tabs is why are some of these values that should always be there nullable.